### PR TITLE
added `validate`, `compose` and `extract` CLI commands

### PR DESCRIPTION
**Pull request by duckki**
- updated command arguments to be consistent across all commands

### CODE REVIEW

**Review by lrlna — APPROVED** [review]

File: cli/src/main.rs [hunk 1]
```diff
@@ -110,3 +145,40 @@ fn dot_federated_graph(file_paths: &[PathBuf]) -> Result<(), FederationError> {
...
+        fs::create_dir_all(dest).map_err(|_| SingleFederationError::Internal {
+            message: "Error: directory creation failed".into(),
+        })?;
+        for (name, subgraph) in subgraphs {
+            let subgraph_path = dest.join(format!("{}.graphql", name));
+            fs::write(subgraph_path, subgraph.schema.schema().to_string()).map_err(|_| {
+                SingleFederationError::Internal {
+                    message: "Error: file output failed".into(),
+                }
+            })?;
```
Comment (goto-bus-stop) [inline cli/src/main.rs:174]: We have a constructor function for internal federation errors:
```suggestion
            fs::write(subgraph_path, subgraph.schema.schema().to_string()).map_err(|_| {
                FederationError::internal("Error: file output failed")
            })?;
```

**Review by goto-bus-stop — COMMENTED** [review]
Small comments, nothing blocking. There's a merge conflict due to #244 🙈

File: cli/src/main.rs (hunk 1 above)
Comment (duckki) [reply cli/src/main.rs:174]: Currently, `FederationError::internal` is `pub(crate)` and not visible to CLI. Shall we open it up to public?

**Comment by duckki** [comment]
@goto-bus-stop Thanks for comments. I'll merge this for now. Let's discuss separately whether FederationError::internal should be public or not.

**Pull request merged**